context.js

```diff
@@ -299,10 +300,25 @@ class Context {
       if (tr.datatype === 'command') return undefined // do not add command aliases to argv
       tr.aliases.forEach(alias => {
         this.argv[alias] = tr.value
+        this.knownArgv[alias] = tr.value
       })
     })
   }
 
+  getUnknownArguments () {
+    return this.argv._
+  }
+
+  getUnknownSlurpedOptions () {
+    const known = Object.keys(this.knownArgv)
+    const provided = Object.keys(this.argv)
+    const unknown = provided.filter(key => !known.includes(key))
+
+    return unknown.map(key => {
+      return this.slurped.find(arg => arg.parsed.find(p => p.key === key))
+    })
+  }
```

Comment:
I think this can be reduced/simplified to this:

```suggestion
  getUnknownSlurpedOptions () {
    return Object.keys(this.argv).filter(key => !(key in this.knownArgv)).map(key => {
      return this.slurped.find(arg => arg.parsed.some(p => p.key === key))
    })
  }
```

If you find this more difficult to grok, then let's at least make the following changes:

1. Remove line 313 (we can use `this.knownArgv` without getting an array of its keys)

2. Change line 315 from this:

    ```js
    const unknown = provided.filter(key => !known.includes(key))
    ```

    to this:

    ```js
    const unknown = provided.filter(key => !(key in this.knownArgv))
    ```

3. Replace the nested `.find()` to `.some()` on line 318, changing it from this:

    ```js
    return this.slurped.find(arg => arg.parsed.find(p => p.key === key))
    ```

    to this:

    ```js
    return this.slurped.find(arg => arg.parsed.some(p => p.key === key))
    ```